harness/determined/core/_context.py

```diff
@@ -44,6 +44,7 @@ def __init__(
         searcher: Optional[core.SearcherContext] = None,
         info: Optional[det.ClusterInfo] = None,
         _tensorboard_manager: Optional[tensorboard.TensorboardManager] = None,
+        utils: Optional[core.UtilsContext] = None,
```

Comment:
can we rename this `core_context.experimental` and `core.ExperimentContext`?

I think the name `Utils` communicates effectively no meaning, and also I think that the way it behaves is a bit suboptimal, and I know we're trying to get something out the door for inference, but I don't think it should be added as a public, stable part of the Core API.

Reply:
added